src/default-directory-provider.coffee

```diff
@@ -0,0 +1,42 @@
+{Directory} = require 'pathwatcher'
+fs = require 'fs-plus'
+path = require 'path'
+
+module.exports =
+class DefaultDirectoryProvider
+
+  # Public: Create a Directory that corresponds to the specified URI.
+  #
+  # * `uri` {String} The path to the directory to add.
+  # * `existingPaths` {Array} The existing paths in atom.project.
+  # Returns:
+  # * {Directory} if the given URI is compatible with this provider.
+  # * {string} if the uri maps to a path an existingPath, in which case
+  #     the path in existingPaths is returned.
```

Comment:
The problem is that normalize() was being called on the path before the directory provider was being consulted, which is not appropriate for URIs. It's not clear to me that `Project` should be responsible for knowing when there is already an appropriate root folder.

There's also the special check for `(directoryPath.indexOf(path.join(existingPath, path.sep)) is 0)`, which seems specific to `DefaultDirectoryProvider`. The idea of `contains()` is specific to a `Directory`, which is the thing we're trying to create, so it doesn't exist yet.